R/common-impact-coding.R

```diff
@@ -0,0 +1,179 @@
+#' @references \url{https://dl.acm.org/citation.cfm?id=507538}
+#' @seealso \code{\link{healthcareai}}
+#' @examples
+#' set.seed(10)
+#' dat <- data.frame(Ycontinuous = sample(5:42, size = 14, replace = TRUE),
+#'                   Ycategorical = sample(c("Y", "N"), size = 14, 
+#'                                        replace = TRUE, prob = c(0.45, 0.55)),
+#'                   YCat01 = c(0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0),
+#'                   hcX = sample(letters[1:5], size = 14, replace = TRUE),
+#'                   hcX2 = sample(letters[6:10], size = 14, replace = TRUE),
+#'                   hcX3 = sample(letters[11:15], size = 14, replace = TRUE),
+#'                   Ymulticlass = sample(c("blue", "green", "red"), size = 14, 
+#'                                        replace = TRUE))
+#'                
+#' # Continuous target and one high_cardinality variable
+#' pphcv(df = dat, target = "Ycontinuous", high_card = "hcX", 
+#'       type = "continuous")
+#'           
+#' # Binary target and one high_cardinality variable
+#' pphcv(df = dat, target = "Ycategorical", high_card = "hcX", pos_class = "Y", 
+#'       type = "binary")
+#'           
+#' # Binary target coded as numeric(0, 1) and one high-cardinality variable 
+#' # and specifying the pos_class.
+#' pphcv(df = dat, target = "YCat01", high_card = "hcX", pos_class = 1, 
+#'       type = "binary")
+#'           
+#' # Binary target coded as numeric(0, 1) and one high-cardinality variable and 
+#' # not specifying the pos_class. Use type = "continuous". Should give same
+#' # results as if type is binary and positive class is specified.
+#' pphcv(df = dat, target = "YCat01",  high_card = "hcX", type = "continuous", 
+#'       m = 1)
+#' 
+#' # Multiclass target and one high-cardinality variable
+#' pphcv(df = dat, target = "Ymulticlass", high_card = "hcX", 
+#'       type = "multiclass")
+#'           
+#' # Binary target and a vector of high-cardinality variables.
+#' vec <- c("hcX", "hcX2", "hcX3")
+#' pphcv(df = dat, target = "Ycategorical", high_card = vec, 
+#'           type = "binary", pos_class = "Y")
+#' 
+#' # Multiclass target and a vector of high-cardinality variables and specify
+#' # remove_high_card = TRUE
+#' vec <- c("hcX", "hcX2", "hcX3")
+#' pphcv(df = dat, target = "Ymulticlass", high_card = vec, type = "multiclass", 
+#'       remove_high_card = TRUE)
+pphcv <- function(df, target, high_card, type, pos_class = NULL, m = NULL,
+                  multi_base = NULL, remove_high_card = FALSE) {
+  
+  for (i in high_card) {
+    # For binary, if target is numeric, pos_class must also be numeric.
+    if (type == "binary" && is.numeric(df[[target]]) && 
+        !is.numeric(pos_class)) {
+      stop("If specified binary target is numeric, pos_Class must also be 
+           numeric. You may also use type = 'continuous' with no specified 
+           pos_class")
+    }
+    # For binary, must have a pos_class specified
```

Comment:
Can you add spaces between these sections (ie, before the comment that explains the new section of code)?